cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -525,6 +525,7 @@ public PartitionOutput getPartitionOutput(PartitionKey key) {
       throw new UnsupportedOperationException(
         "Output is not supported for external partitioned file set '" + spec.getName() + "'");
     }
+    validatePartitionKey(key, partitioning);
     return new BasicPartitionOutput(this, getOutputPath(key), key);
```

Comment:
I did not add a check there because it is not a public method. But you are right, it could get called from PartitionedFileSetDefinition because it is package visible. Fixed.